drivers/interrupt_controller/intc_nuclei_eclic.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nuclie's Extended Core Interrupt Controller
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <sys/util.h>
+#include <init.h>
+#include <soc.h>
+
+#include <sw_isr_table.h>
+
+union CLICCFG {
+	struct {
+		uint8_t _reserved0 : 1;
+		uint8_t nlbits : 4;        /*!< number of interrupt level bits */
+		uint8_t _reserved1 : 2;
+		uint8_t _reserved2 : 1;
+	} b;
+	uint8_t w;
+};
+
+union CLICINFO {
+	struct {
+		uint32_t numint : 13;           /*!< number of max supported interrupts */
+		uint32_t version : 8;           /*!< architecture version */
+		uint32_t intctlbits : 4;        /*!< supported bits in the clicintctl */
+		uint32_t _reserved0 : 7;
+	} b;
+	uint32_t qw;
+};
+
+union CLICMTH {
+	uint8_t w;
+};
+
+union CLICINTIP {
+	struct {
+		uint8_t IP : 1;  /*!< Interrupt Pending */
+		uint8_t reserved0 : 7;
+	} b;
+	uint8_t w;
+};
+
+union CLICINTIE {
+	struct {
+		uint8_t IE : 1;  /*!< Interrupt Enabled */
+		uint8_t reserved0 : 7;
+	} b;
+	uint8_t w;
+};
+
+union CLICINTATTR {
+	struct {
+		uint8_t shv : 1;        /*!< 0: non-vectored 1:vectored */
+		uint8_t trg : 2;        /*!< 0: level 1: rising edge 2: falling edge */
+		uint8_t reserved0 : 3;
+		uint8_t reserved1 : 2;
+	} b;
+	uint8_t w;
+};
+
+struct CLICCTRL {
+	volatile union CLICINTIP INTIP;
+	volatile union CLICINTIE INTIE;
+	volatile union CLICINTATTR INTATTR;
+	volatile uint8_t INTCTRL;
+};
+
+#define ECLIC_MODE_MTVEC_Msk   3U      /*!< ECLIC Mode mask for MTVT CSR Register */
+
+#define CLIC_INTATTR_TRIG_Pos  1U      /*!< CLIC INTATTR: TRIG Position */
+#define CLIC_INTATTR_TRIG_Msk  (0x3UL << CLIC_INTATTR_TRIG_Pos)    /*!< CLIC INTATTR: TRIG Mask */
+
+#define ECLIC_CFG    (*((volatile union CLICCFG *)(DT_REG_ADDR_BY_IDX(DT_NODELABEL(eclic), 0))))
+#define ECLIC_INFO  (*((volatile union CLICINFO *)(DT_REG_ADDR_BY_IDX(DT_NODELABEL(eclic), 1))))
+#define ECLIC_MTH    (*((volatile union CLICMTH *)(DT_REG_ADDR_BY_IDX(DT_NODELABEL(eclic), 2))))
+#define ECLIC_CTRL   ((volatile struct CLICCTRL *)(DT_REG_ADDR_BY_IDX(DT_NODELABEL(eclic), 3)))
+#define ECLIC_CTRL_SIZE                (DT_REG_SIZE_BY_IDX(DT_NODELABEL(eclic), 3))
+
+#if CONFIG_3RD_LEVEL_INTERRUPTS
+#define INTERRUPT_LEVEL 2
+#elif CONFIG_2ND_LEVEL_INTERRUPTS
+#define INTERRUPT_LEVEL 1
+#else
+#define INTERRUPT_LEVEL 0
+#endif
+
+static uint8_t NLBITS;
+static uint8_t INTCTLBITS;
+static uint8_t MAX_PRIO;
+static uint8_t MAX_LEVEL;
+static uint8_t INTCTRL_MASK;
+
+static inline uint8_t LALIGN8(uint8_t val, int shift)
```

Comment:
use macros if you prefer `UPPERCASE_HELPERS`

Reply:
Changing function name to lower case.